pkg/cloud/azureprovider.go

```diff
@@ -734,6 +828,49 @@ func (az *Azure) AllNodePricing() (interface{}, error) {
 func (az *Azure) NodePricing(key Key) (*Node, error) {
 	az.DownloadPricingDataLock.RLock()
 	defer az.DownloadPricingDataLock.RUnlock()
+	config, _ := az.Config.GetCustomPricingData()
+	if config.SpotLabel != "" && config.SpotLabelValue != "" {
+		labelsMapStr := fmt.Sprintf("%s", key)
+		spotLabelMapStr := " " + config.SpotLabel + ":" + config.SpotLabelValue + " "
+		if strings.Contains(labelsMapStr, spotLabelMapStr) {
+			features := strings.Split(key.Features(), ",")
+			spotFeatures := strings.ReplaceAll(key.Features(), ",ondemand", ",spot")
+			if n, ok := az.Pricing[spotFeatures]; ok {
```

Comment:
I don't know, there might be a chance that MS will add spot type to their ratecard, I've seen it somewhere within the issues on their github. But I can just remove it.

Reply:
Ok, seems like good defensive coding then, ideally it would be best if the spot was populated from the ratecard api in the future. Maybe just add a small comment to it.